pkg/controller/acmechallenges/sync.go

```diff
@@ -38,7 +38,11 @@ import (
 )
 
 const (
-	reasonDomainVerified = "DomainVerified"
+	reasonDomainVerified  = "DomainVerified"
+	challengeCleanUpError = "CleanUpError"
+	challengePresentError = "PresentError"
+	statusPresented       = "Presented"
+	statusFailed          = "Failed"
```

Comment:
```suggestion
	reasonChallengeCleanUpError = "CleanUpError"
	reasonChallengePresentError = "PresentError"
	reasonStatusPresented       = "Presented"
	reasonStatusFailed          = "Failed"
```

Is what I meant. i.e. they all need the `reason???` prefix for consistency with the existing `reasonDomainVerified`.

Reply:
If it helps, here is an example of how we do it elsewhere https://github.com/jetstack/cert-manager/blob/master/pkg/issuer/acme/setup.go#L44 . I think the prefix makes it easier to understand what the constant is (i.e error, reason, message etc) not just the content of it.

Reply:
Yeah that makes sense @irbekrm - that's why I used the `status` prefix for "Presented" and "Failed" instead of the `reason` prefix. Would that be the goal? Or should I just go for a common prefix for all the constants in a file? 🙂

Reply:
All the constants in all the files in this PR are used as `reason` argument in calls to the events `Eventf` API. 
So add the `reason???` prefix to all of them.

It makes it easier to grep for and it makes the purpose of the constants clearer.

In contrast, Kubernetes Event reason constants use a suffix, `FooBarReason`  which is harder to grep for. e.g.


```
[richard@drax kubernetes]$ git grep '[a-z]Reason = "'
...

pkg/controller/controller_utils.go:     FailedCreatePodReason = "FailedCreate"
pkg/controller/controller_utils.go:     SuccessfulCreatePodReason = "SuccessfulCreate"
pkg/controller/controller_utils.go:     FailedDeletePodReason = "FailedDelete"
pkg/controller/controller_utils.go:     SuccessfulDeletePodReason = "SuccessfulDelete"
pkg/controller/daemon/daemon_controller.go:     SelectingAllReason = "SelectingAll"
pkg/controller/daemon/daemon_controller.go:     FailedPlacementReason = "FailedPlacement"
pkg/controller/daemon/daemon_controller.go:     FailedDaemonPodReason = "FailedDaemonPod"
pkg/controller/deployment/util/deployment_util.go:      ReplicaSetUpdatedReason = "ReplicaSetUpdated"
pkg/controller/deployment/util/deployment_util.go:      FailedRSCreateReason = "ReplicaSetCreateError"
pkg/controller/deployment/util/deployment_util.go:      NewReplicaSetReason = "NewReplicaSetCreated"
pkg/controller/deployment/util/deployment_util.go:      NewRSAvailableReason = "NewReplicaSetAvailable"
pkg/controller/deployment/util/deployment_util.go:      TimedOutReason = "ProgressDeadlineExceeded"
pkg/controller/deployment/util/deployment_util.go:      PausedDeployReason = "DeploymentPaused"
pkg/controller/deployment/util/deployment_util.go:      ResumedDeployReason = "DeploymentResumed"
pkg/controller/job/job_controller.go:           failureReason = "BackoffLimitExceeded"
pkg/controller/job/job_controller.go:           failureReason = "DeadlineExceeded"
pkg/controller/podautoscaler/horizontal.go:             rescaleReason = "Current number of replicas above Spec.MaxReplicas"
pkg/controller/podautoscaler/horizontal.go:             rescaleReason = "Current number of replicas below Spec.MinReplicas"
pkg/controller/podautoscaler/horizontal.go:                     rescaleReason = "All metrics below target"
pkg/controller/podautoscaler/horizontal.go:                     possibleLimitingReason = "ScaleUpLimit"
pkg/controller/podautoscaler/horizontal.go:                     possibleLimitingReason = "TooManyReplicas"
pkg/controller/podautoscaler/horizontal.go:                     possibleLimitingReason = "ScaleDownLimit"
pkg/controller/podautoscaler/horizontal.go:                     possibleLimitingReason = "TooFewReplicas"

```